analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AssertionsShouldBeComplete.FluentAssertions.CSharp7.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using FluentAssertions.Primitives;
+
+public class Program
+{
+    public void StringAssertions()
+    {
+        var s = "Test";
+        s.Should();                   // Noncompliant {{Complete the assertion}}
+        //^^^^^^
+        s[0].Should();                // Error [CS0121] ambiguous calls
+                                      // Noncompliant@-1
+
+        s.Should().Be("Test");        // Compliant
+    }
+
+    public void CollectionAssertions()
+    {
+        var collection = new[] { "Test", "Test" };
+        collection.Should();                               // Error [CS0121] ambiguous calls
+                                                           // Noncompliant@-1
+        collection.Should<string>();                       // Noncompliant
+
+        collection.Should<string>().Equal("Test", "Test"); // Compliant
+    }
+
+    public void DictionaryAssertions()
+    {
+        var dict = new Dictionary<string, object>();
+        dict["A"].Should();    // Noncompliant
+    }
+
+    public StringAssertions ReturnedByReturn()
+    {
+        var s = "Test";
+        return s.Should();     // Compliant
+    }
```

Comment:
```suggestion
    public StringAssertions ReturnedByReturn()
    {
        var s = "Test";
        s.Should();            // Noncompliant
        return s.Should();     // Compliant
    }
```

Reply:
We test that case already in line 11.

Reply:
Not in a method with a return value. But you're right, this might be unnecessary.